modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ButtonTag.java

```diff
@@ -254,6 +254,15 @@ protected void cleanUp() {
 		_type = "button";
 	}
 
+	@Override
+	protected String getHydratedModuleName() {
+		if ((getAdditionalProps() != null) && (getPropsTransformer() != null)) {
```

Comment:
Do we want the component not to be hydrated at least both of this props are passed? Could it be the case in which I want to pass a `propsTransformer` but no `additionalProps` or viceversa?

Reply:
Also I think this check is usually done with `Validator.isNotNull()`

Reply:
> Also I think this check is usually done with Validator.isNotNull()

We use `Validator.isNotNull()` when checking for null-ish values, specially in strings or numbers. That is, empty string returns `true` for `Validator.isNotNull('')`. Strict comparisons to `null` are preferred given the proper types.

> Do we want the component not to be hydrated at least both of this props are passed? Could it be the case in which I want to pass a propsTransformer but no additionalProps or viceversa?

Yeah, should be one or the other, not both.

Reply:
> We use Validator.isNotNull() when checking for null-ish values, specially in strings or numbers. That is, empty string returns true for Validator.isNotNull(''). Strict comparisons to null are preferred given the proper types.

Good to know! I thought we were using it always, thx!

Reply:
> We use Validator.isNotNull() when checking for null-ish values, specially in strings or numbers. That is, empty string returns true for Validator.isNotNull(''). Strict comparisons to null are preferred given the proper types.

Sorry, I don't understand, if I should `Validator.isNull()` or a strict comparison 

Since `propsTransfomer` is a `String` and (I'm guessing) someone could eventually do

```js
<clay:button
    propsTransformer=""
/>
```

That makes me think that I should use 

```java
if (Validator.isNotNull(propsTransformer)) {
 ...
}
```

Is that correct @jbalsas?

If that's the case, is there also a similar pattern for `Map` or `HashMap`, because that's the type used for `additionalProps`

Reply:
Just another question @jbalsas: 

I don't understand why someone would use `additionalProps` without a `propsTransformer`: because if you pass `additionalProps` it's probably to access/use them later and as far as I understand if you don't pass a `propsTransformer` those `additionalProps` are going to be passed to the `@clayui` component which isn't going to do anything with them ... or could it be that you want to pass props to the `@clayu*` component that the tag doesn't support and so you use `additionalProps` so that the component receive them?